Integrations/integration-Vectra.yml

```diff
@@ -0,0 +1,229 @@
+description: Uses Vectra end point security ESM
+configuration:
+- display: Server URL (e.g. https://192.168.0.1)
+  name: server
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Credentials
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Fetch incidents
+  name: isFetch
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |-
+    var username = params.credentials.identifier;
+    var password = params.credentials.password;
+    var server = params.server;
+    var insecure = params.insecure;
+
+    var sendRequest = function(method, url, queryName) {
+        var res = http(
+                url,
+                {
+                    Method: method,
+                    Username: username,
+                    Password: password,
+                    Headers: {'content-type': ['application/json']},
+                },
+                insecure
+            );
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Failed to ' + queryName + ', request status code: ' + res.StatusCode + ' and Body: ' + res.Body + '.';
+        }
+        return JSON.parse(res.Body);
+    }
+
+    var urlDict = {
+        'vec-detections': '/api/detections',
+        'vec-hosts':'/api/hosts',
+        'vec-settings':'/api/settings',
+        'vec-triage':'/api/rules',
+        'vec-sensors':'/api/sensors',
+    }
+
+    var createIncidentFromDetection = function(incident) {
+        var keys = Object.keys(incident);
+        var labels = [];
+        for (var i = 0; i<keys.length; i++) {
+            labels.push({'type': keys[i], 'value': typeof(incident[keys[i]]) === 'object' ? JSON.stringify(incident[keys[i]]) : incident[keys[i]].toString()});
+        }
+        return {
+            "name": incident.id + ' ' + incident['type_vname'],
+            "labels": labels,
+        }
+    }
+
+    switch (command) {
+        case 'fetch-incidents':
+            var lastRun = getLastRun();
+            var id = lastRun && lastRun.id ? lastRun.id : 0;
+            var res = sendRequest('GET', server + urlDict['vec-detections'] + encodeToURLQuery({min_id: id, 'page':'last'}), 'vec-detections').results;
+            var incidents = [];
+            for (var i = 0; i < res.length; i++) {
+                var detection = res[i];
+                id = Math.max(id, detection.id);
```

Comment:
Are you sure that IDs will be asc ?
Can it be that an older ID will have a newer timestamp ?
Ids would be reused ?